loyalty_partner_applicability/models/res_config_settings.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2023 Tecnativa - Pilar Vargas Moreno
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+
+    _inherit = "res.config.settings"
+
+    allow_coupon_sharing = fields.Boolean(
+        string="Allow coupon sharing", config_parameter="allow_coupon_sharing"
```

Comment:
Add a help string describing the effect of this: if checked, all the commercial entity members of the partner will fullfil the rule... (or something along the lines)

Reply:
Yes, the field is added in the extended view "sale.res_config_settings_view_form" and that is why it is done in the module sale_*, in the section "Pricing". 
https://github.com/OCA/sale-promotion/blob/1331bce2984e48c8ce0555347ad3d03ee98897ec/sale_loyalty_partner_applicability/views/res_config_settings.xml#L5-L28
I had added a title as a help as they do in odoo but if it is more recommendable I add the help in the field itself.

Reply:
Ok :) Add comment so there's no doubt in the future :D (the help string can be that one)

Reply:
done! :)